core/src/main/scala/akka/persistence/r2dbc/internal/Sql.scala

```diff
@@ -81,4 +82,29 @@ object Sql {
     }
   }
 
+  final class Cache {
```

Comment:
There is also this `mutable.Map.getOrElseUpdate` from [here](https://www.scala-lang.org/api/2.13.7/scala/collection/mutable/Map.html#getOrElseUpdate(key:K,op:=%3EV):V) - maybe this is more convenient, but I am not sure about the performance implications.

Reply:
I think we can use immutable here since there will be no updates after initial population